specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingProperty.json

```diff
@@ -65,182 +95,553 @@
             "$ref": "./types.json#/parameters/apiVersionParameter"
           },
           {
-            "description": "Request parameters that are provided to the update billing property operation.",
             "in": "body",
-            "name": "parameters",
-            "required": true,
+            "name": "body",
+            "description": "A billing property.",
             "schema": {
               "$ref": "#/definitions/BillingProperty"
             }
           }
         ],
         "responses": {
           "200": {
-            "description": "OK. The request has succeeded.",
+            "description": "A billing property.",
             "schema": {
               "$ref": "#/definitions/BillingProperty"
             }
           },
           "default": {
-            "description": "Error response describing why the operation failed.",
+            "description": "Default response",
             "schema": {
-              "$ref": "./types.json#/definitions/ArmError"
+              "$ref": "./types.json#/definitions/ArmErrorResponse"
             }
           }
         },
-        "tags": [
-          "BillingProperties"
-        ],
         "x-ms-examples": {
-          "billingPropertiesPatch": {
-            "$ref": "./examples/billingPropertiesPatch.json"
+          "BillingPropertyPatchCostCenter": {
+            "$ref": "./examples/billingPropertyPatchCostCenter.json"
+          },
+          "BillingPropertyPatchSubscriptionServiceUsageAddress": {
+            "$ref": "./examples/billingPropertyPatchSubscriptionServiceUsageAddress.json"
           }
         }
       }
     }
   },
   "definitions": {
     "BillingProperty": {
+      "description": "A billing property.",
+      "type": "object",
       "allOf": [
         {
           "$ref": "./types.json#/definitions/ArmResource"
         }
       ],
-      "description": "A billing property.",
       "properties": {
         "properties": {
-          "$ref": "#/definitions/BillingPropertyProperties",
-          "description": "A billing property.",
+          "description": "The properties of a(n) BillingProperty",
+          "allOf": [
+            {
+              "$ref": "#/definitions/BillingPropertyProperties"
+            }
+          ],
           "x-ms-client-flatten": true
         }
       },
-      "type": "object"
+      "x-ms-azure-resource": true
     },
     "BillingPropertyProperties": {
-      "description": "The billing property.",
+      "description": "A billing property.",
+      "type": "object",
       "properties": {
```

Comment:
Have we considered grouping properties together? E.g.:

```
"billingAccount": {
  "agreementType": "...",
  "status": "...",
  etc.
},
"billingProfile": {
  ...
}
```

Reply:
@ramkd12 @amberbhargava to comment here